src/types.ml

```diff
@@ -528,8 +528,14 @@ struct
 
     (* TODO: when offset is 0 then do not print anything, if offset is N (memargN) then print nothing ? *)
     let memarg fmt { offset; align } =
+      let rec pow_2 n =
```

Comment:
As we're only computing `pow_2 x` and not `pow x n` we can do something better than the recursive algorithm. Hint: think about how bits (base 2) and power of 2 are related. You can do it in `O(1)`.

Reply:
`lsl` operator ;)